modelforge/potential/models.py

```diff
@@ -686,124 +682,180 @@ def _input_checks(self, data: Union[NNPInput, NamedTuple]):
                     )
-                    stddev = unit.Quantity(atomic_energies_stats[proc["stddev"]]).m_as(
-                        unit.kilojoule_per_mole
+                    prostprocessing_sequence_names.append("normalize")
+                # check if also reduction is requested
+                if operations.get("from_atom_to_molecule_reduction", False):
+                    postprocessing_sequence.append(
+                        FromAtomToMoleculeReduction(
+                            per_atom_property_name="per_atom_energy",
+                            index_name="atomic_subsystem_indices",
+                            output_name="per_molecule_energy",
+                            keep_per_atom_property=operations.get(
+                                "keep_per_atom_property", False
+                            ),
+                        )
                     )
-                operation = ScaleValues(mean=mean, stddev=stddev)
-                work_to_be_done_per_property.append(operation)
-                props.append(proc)
-
-            elif proc["step"] == "calculate_ase":
-                if dataset_statistic is None:
-                    raise RuntimeError(
-                        "No dataset statistics provided for ASE calculation. Skipping!"
+                    prostprocessing_sequence_names.append(
+                        "from_atom_to_molecule_reduction"
                     )
-                else:
-                    atomic_self_energies = dataset_statistic["atomic_self_energies"]
-                    operation = CalculateAtomicSelfEnergy(atomic_self_energies)
-                    work_to_be_done_per_property.append(operation)
-
-                props.append(proc)
-
-        self.per_atom_operations = ModuleList(work_to_be_done_per_property)
-        self.per_atom_operations_prop = props
-
-        # initialize per molecule reduction
-        work_to_be_done_per_property = []
-        props = []
-        for proc in readout_operation:
-            if proc["step"] == "from_atom_to_molecule":
-                operation = FromAtomToMoleculeReduction(
-                    reduction_mode=proc["mode"],
-                )
-            work_to_be_done_per_property.append(operation)
-            props.append(proc)
+            elif property == "general_postprocessing_operation":
+                # check if also self-energies are requested
+                if operations.get("calculate_molecular_self_energy", False):
 
-        self.readout_operation = ModuleList(work_to_be_done_per_property)
-        self.readout_prop = props
+                    if self.dataset_statistic is None:
+                        log.warning(
+                            "Dataset statistics are required to calculate the molecular self-energies but haven't been provided."
+                        )
+                    else:
+                        atomic_self_energies = self.dataset_statistic[
+                            "atomic_self_energies"
```

Comment:
Again, not sure if we wanted this to follow a specific naming convention: "energy-atomic_self" (although that doesn't as cleanly fit into that schema).